--- venv/lib/python3.12/site-packages/vllm/platforms/interface.py.original	2025-10-27 20:47:23.575321655 -0700
+++ venv/lib/python3.12/site-packages/vllm/platforms/interface.py	2025-10-27 20:57:18.998148656 -0700
@@ -35,8 +35,8 @@
 
 def in_wsl() -> bool:
     # Reference: https://github.com/microsoft/WSL/issues/4071
-    return "microsoft" in " ".join(uname()).lower()
-
+    uname_str = " ".join(uname()).lower()
+    return "microsoft" in uname_str and not "wsl2" in uname_str
 
 class _Backend(enum.Enum):
     FLASH_ATTN = enum.auto()
